File: Additional Example/Custom Request - Edit Header/Python/encrypt.py
import argparse
from base64 import b64decode,b64encode
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.primitives.padding import PKCS7
from cryptography.hazmat.backends import default_backend
import json
import hashlib

# Create an argument parser
parser = argparse.ArgumentParser(description='Process data argument')
parser.add_argument('-d', '--data', help='File path with encrypted data + base64 in JSON format')

# Parse the arguments
args = parser.parse_args()

def aes_cbc_encrypt(plaintext, key, iv):
    # Convert the key and IV to bytes
    key = key.encode('utf-8')
    iv = iv.encode('utf-8')

    # Create a Cipher object with AES CBC algorithm and PKCS7 padding
    cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend())
    encryptor = cipher.encryptor()

    # Pad the plaintext using PKCS7
    padder = PKCS7(algorithms.AES.block_size).padder()
    padded_plaintext = padder.update(plaintext.encode('utf-8')) + padder.finalize()

    # Encrypt the padded plaintext
    ciphertext = encryptor.update(padded_plaintext) + encryptor.finalize()

    # Base64 encode the ciphertext
    encoded_ciphertext = b64encode(ciphertext).decode('utf-8')

    # Return the encoded ciphertext
    return encoded_ciphertext


def update_signature(decoded_header,updatedbody):

    headersplit = decoded_header.split("\n")
    md5_hash = hashlib.md5(updatedbody.encode('utf-8')).hexdigest()

    for i in range(len(headersplit)):
        if headersplit[i].startswith('Signature:'):
            
            headersplit[i] = f'Signature: {md5_hash}'
            break
    updated_header = "\n".join(headersplit)
    return updated_header



with open(args.data, 'r') as file:    
    content_body = json.load(file).get("data")
    content_header = json.load(file).get("header")
        
jsondata = b64decode(content_body).decode('utf-8')
decoded_header = b64decode(content_header).decode('utf-8')


data = json.loads(jsondata)
ciphertext = data['user_id']
key = "mysecretkey12345"
iv = "n2r5u8x/A%D*G-Ka"
encrrypted = aes_cbc_encrypt(ciphertext, key, iv)
data['user_id'] = encrrypted
updatedbody = json.dumps(data)




updated_header = update_signature(decoded_header,updatedbody)


# Print the updated header
print(b64encode(updated_header.encode('utf-8')).decode('utf-8'))
print(b64encode(updatedbody.encode('utf-8')).decode('utf-8'))

File: Additional Example/Custom Request - Edit Header/JS/encrypt.js
const fs = require('fs');
const path = require('path');
var CryptoJS = require("crypto-js");
const { program } = require('commander');
const { Buffer } = require('buffer');

program
  .option('-d, --data <file_path>', 'Path to JSON file containing base64 encoded + encrypted data');
  
program.parse(process.argv);
const options = program.opts();
 
const filePath = options.data;
const absoluteFilePath = path.resolve(filePath);
var data = fs.readFileSync(absoluteFilePath).toString();
const jsonData = JSON.parse(data);
const body = Buffer.from(jsonData.data, 'base64').toString('utf8');
var header = Buffer.from(jsonData.header, 'base64').toString('utf8');



var parsedata = JSON.parse(body)


// go through json and encrypt each json value
for (var key in parsedata) {
    // Check if the property exists in the object (not inherited)
    if (parsedata.hasOwnProperty(key)) {
        // Update the value of each property
        parsedata[key] = encrypt(parsedata[key]);
    }
}

// updated json body
var updatedbody = JSON.stringify(parsedata);


// encryption code
function encrypt(value) {
    var key = "mysecretkey12345"
    var iv = "n2r5u8x/A%D*G-Ka"
    var bytes = CryptoJS.AES.encrypt(value, CryptoJS.enc.Utf8.parse(key), {
        keySize: 128 / 8,
        iv: CryptoJS.enc.Utf8.parse(iv),
        mode: CryptoJS.mode.CBC
    });
    var originalText = bytes.toString();
    return originalText
}



// split the raw http request header 
var headersplit = header.split(/\r?\n/)

// go through json and update the signature header with updated body md5 hash
for (var i = 0; i < headersplit.length; i++) {
    if (headersplit[i].startsWith('Signature:')) {
        headersplit[i] = 'Signature: ' + CryptoJS.MD5(updatedbody).toString();
        break;
    }
}

// join the header to as original format
var updatedheader = headersplit.join("\r\n")

// printt the updated base64 header and updated base64 body
console.log(Buffer.from(updatedheader).toString('base64'));
console.log(Buffer.from(updatedbody).toString('base64'));